packages/datatrak-web/src/features/Survey/SurveyContext/SurveyContext.tsx

```diff
@@ -37,11 +42,12 @@ export const SurveyContext = ({ children }) => {
   const { surveyCode, ...params } = useParams<SurveyParams>();
   const screenNumber = params.screenNumber ? parseInt(params.screenNumber!, 10) : null;
   const { data: survey } = useSurvey(surveyCode);
+  const isResponseScreen = !!useMatch(ROUTES.SURVEY_RESPONSE);
 
   const { formData } = state;
 
   const surveyScreens = survey?.screens || [];
-  const flattenedScreenComponents = getAllSurveyComponents(surveyScreens);
+  const flattenedScreenComponents = getAllSurveyComponents(surveyScreens) ?? [];
```

Comment:
Would it make more sense to fall back to the empty array in the function definition?

```ts
/* utils.ts */
export const getAllSurveyComponents = (surveyScreens?: SurveyScreen[]) => {
  if (!surveyScreens) return [];
  // Existing stuff
}
```

I’m not 100% sure what `SurveyScreen` is, so possible this wouldn’t make semantic sense